src/features/profile/helpers/getEmailUpdateStep.ts

```diff
@@ -1,13 +1,16 @@
 import { EmailHistoryEventTypeEnum } from 'api/gen'
 
-export function getEmailUpdateStep(step?: EmailHistoryEventTypeEnum) {
+export function getEmailUpdateStep(
+  hasRecentlyResetPassword: boolean,
+  step?: EmailHistoryEventTypeEnum
+) {
   switch (step) {
-    case EmailHistoryEventTypeEnum.UPDATE_REQUEST:
-      return 1
     case EmailHistoryEventTypeEnum.CONFIRMATION:
-      return 2
+      return 1 + Number(hasRecentlyResetPassword)
+    case EmailHistoryEventTypeEnum.NEW_EMAIL_SELECTION:
+      return 2 + Number(hasRecentlyResetPassword)
     case EmailHistoryEventTypeEnum.VALIDATION:
-      return 3
+      return 3 + Number(hasRecentlyResetPassword)
```

Comment:
DRY

```suggestion
  const toto = hasRecentlyResetPassword ? 1 : 0
  switch (step) {
    case EmailHistoryEventTypeEnum.CONFIRMATION:
      return 1 + toto
    case EmailHistoryEventTypeEnum.NEW_EMAIL_SELECTION:
      return 2 + toto
    case EmailHistoryEventTypeEnum.VALIDATION:
      return 3 + toto
```

je suis pas inspiré pour toto